include/pybind11/detail/type_caster_base.h

```diff
@@ -38,6 +38,7 @@ class loader_life_support {
     }
 
     /// ... and destroyed after it returns
+    // NOLINTNEXTLINE(bugprone-exception-escape) See issue #2215
```

Comment:
This is my only strong suggestion:
IIUC we're NOT turning on bugprone-exception-escape in this PR. Is that true?
In that case it will be confusing to have the NOLINT in this PR, could you please remove them?
I realize it's a little bit of extra work adding them back in later, but IMM making this PR as straightforward/clean as possible is an important enough reason to invest those 5 minutes.